src/client.ts

```diff
@@ -549,26 +548,26 @@ export class Client {
 
   private async getConnection(
     streamName: string,
-    leader: boolean,
+    purpose: "publisher" | "consumer",
```

Comment:
I think it would make sense to define a union type and a couple of constants to reduce the use of plain strings